mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/dto/query/QueryElementSerializerTest.java

```diff
@@ -0,0 +1,90 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault.dto.query;
+
+import static org.apache.james.vault.DeletedMessageFixture.SUBJECT;
+import static org.apache.mailet.base.MailAddressFixture.SENDER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.james.vault.dto.query.QueryTranslator.FieldName;
+import org.apache.james.vault.dto.query.QueryTranslator.Operator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.javacrumbs.jsonunit.assertj.JsonAssertions;
+
+class QueryElementSerializerTest {
+
+    private QueryElementSerializer queryElementSerializer;
+
+    @BeforeEach
+    void beforeEach() {
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
it's usually a better idea to use an ObjectMapper configured by a production component because the configuration matters (a lot)

Reply:
Which component do you have in mind ? I have seen the other unit test using the objectMapper and they used a similar approach as this one.

Reply:
I think that we can have a single ObjectMapper configuration for all task serialization. We can define this ObjectMapper in a way that make it simple to inject in guice and at the same time easy to import in tests

Reply:
All the tasks use the same object mapper which is a private field created in the constructor of JsonGenericSerializer which is not exposed itself in JsonTaskSerializer. Gautier is doing a PR in this direction. And this change could be included in it #2553